cpp/benchmarks/CMakeLists.txt

```diff
@@ -1,6 +1,18 @@
-cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
-
-project(CUDF_BENCHS LANGUAGES C CXX CUDA)
```

Comment:
This removes the project -- that usually causes a cmake warning in my experience.

Reply:
I believe @jrhemstad and I tested this in other projects and for cases where the `CMakeLists.txt` is only added via a `add_subdirectory` call it's safe to not have a `project` command.

Reply:
I don't think so. Check the CI output of this PR. I see 

```
CMake Warning (dev) in CMakeLists.txt:
  No project() command is present.  The top-level CMakeLists.txt file must
  contain a literal, direct call to the project() command.  Add a line of
  code such as

    project(ProjectName)

  near the top of the file, but after cmake_minimum_required().

  CMake is pretending there is a "project(Project)" command on the first
  line.
This warning is for project developers.  Use -Wno-dev to suppress it.
```

I also see this because you removed `cmake_minimum_required`.

```
CMake Warning (dev) in CMakeLists.txt:
  No cmake_minimum_required command is present.  A line of code such as

    cmake_minimum_required(VERSION 3.17)

  should be added at the top of the file.  The version specified may be lower
  if you wish to support older CMake versions for this project.  For more
  information run "cmake --help-policy CMP0000".
This warning is for project developers.  Use -Wno-dev to suppress it.
```

Reply:
Looks like both of these are from how we're fetching GoogleTest as opposed to the `tests/CMakeLists.txt` file itself. These will be removed with #6350 anyway, so I'll add the `project` command back in to alleviate these warnings.